articles/tutorials/advanced/2d_shaders/05_transition_effect/index.md

```diff
@@ -0,0 +1,528 @@
+```hlsl
+#if OPENGL  
+    #define SV_POSITION POSITION  
+    #define VS_SHADERMODEL vs_3_0  
+    #define PS_SHADERMODEL ps_3_0  
+#else  
+    #define VS_SHADERMODEL vs_4_0_level_9_1  
+    #define PS_SHADERMODEL ps_4_0_level_9_1  
+#endif  
+  
+Texture2D SpriteTexture;  
+  
+float Progress;  
+  
+sampler2D SpriteTextureSampler = sampler_state  
+{  
+    Texture = <SpriteTexture>;  
+};  
+  
+struct VertexShaderOutput  
+{  
+    float4 Position : SV_POSITION;  
+    float4 Color : COLOR0;  
+    float2 TextureCoordinates : TEXCOORD0;  
+};  
+  
+float4 MainPS(VertexShaderOutput input) : COLOR  
+{  
+    return float4(Progress, 0, 0, 1);  
+}  
+  
+  
+technique SpriteDrawing  
+{  
+    pass P0  
+    {  
+        PixelShader = compile PS_SHADERMODEL MainPS();  
+    }  
+};
+```
+
+Now you can use the slider in the debug UI to visualize the `Progress` parameter as the red channel. 
+![Figure 5.3: See the Progress parameter in the red value](./gifs/progress-parameter.gif)
+
+### Coordinate Space
+
+If we are making a screen wipe, then parts of the screen will be transitioning before other parts of the screen. The shader will affect the screen based on the coordinate of the pixel on the screen. For example, if we wanted to make a horizontal screen wipe, we would need to know the x-coordinate of each pixel. With the x-coordinate of a pixel, the shader could decide if that pixel should be shown or hidden based on the transition's progress parameter. 
+
+The shader actually provides the x-coordinate of each pixel in the `input.TextureCoordinates` structure. 
+
+>[!TIP] 
+> `input.TextureCoordinates` are not _actually_ the pixel coordinates
+> 
+> In this example, the `input.TextureCoordinates` represents pixel coordinates _because_ the sprite is being drawn as a full screen quad. However, if the sprite was not taking up the entire screen, the texture coordinates would behave differently. This topic will be discussed more later on.
+
+The following shader helps visualize the x-coordinate of each pixel. 
+```hlsl
+float2 uv = input.TextureCoordinates;  
+return float4(uv.x, 0, 0, 1);
+```
+
+That results in this image, where the left edge has an x-coordinate of `0`, it has no red value, and where the right edge has an x-coordinate of `1`, the image is fully red. In the middle of the image, the red value interpolates between `0` and `1`. 
+![Figure 5.4: the x coordinate of each pixel represented in the red channel](./images/x-pos.png)
+
+The same pattern holds true for the y-coordinate. Observe the following shader code putting the y-coordinate of each pixel in the green channel.
+
+```hlsl
+float2 uv = input.TextureCoordinates;  
+return float4(0, uv.y, 0, 1);
+```
+
+As you can see, the top of the screen has a `0` value for the y-coordinate, and the bottom of the screen has a `1`. 
+
+![Figure 5.5: The y-coordinate of each pixel represented in the green channel](./images/y-pos.png)
```

Comment:
This screenshot confirms the previous statement about IMGUI, without changing any implementation about the "Progress" parameter added earlier, the IMGUI screen has changed back to its original state, should a slider not be present if nothing has changed?